README.md
# flutter_string_converter

This Flutter Package for converting string to date, int, double or set as you want. But the only condition is that the string is in format of that convertible type

# Using
## What can you do with this package.

- To first letter of string capital example
 ``` 
 "your string".toFirstCapital() 
 ```
and you get result Your string
- To all first letter of string capital example
 ``` 
 "your string".toEveryFirstCapital() 
 ```
and you get result Your String
- To set in camel case of string example
 ``` 
 "your string".toCamelCase() 
 ```
and you get result yOuR StRiNg
- To get only first character of string example
 ``` 
 "Your string".toSingleLetter() 
 ```
and you get result Y
- To get all first letter of string with joinWith example
 ``` 
 "your string".toEverySingleLetter(joinWith: ".") 
 ```
and you get result Y.S
- To get double value from string example
 ``` 
 "1.2".toDouble() 
 ```
- To get int value from string example
 ``` 
 "1".toInt() 
 ```
- To get date with any format you want example
 ``` 
 "2021-05-10".toDate(format: FormatDate.dmy) 
 ```
and you get result 10/05/2021

## Date Format

| Format Name | Result |
| --- | --- |
| FormatDate.dmyDash | 16-08-2021 |
| FormatDate.mdyDash | 08-16-2021 |
| FormatDate.ymdDash | 2021-08-16 |
| FormatDate.ydmDash | 2021-16-08 |
| FormatDate.dmy | 16/08/2021 |
| FormatDate.mdy | 08/16/2021 |
| FormatDate.ymd | 2021/08/16 |
| FormatDate.ydm | 2021/16/08 |
| FormatDate.dmyDash12 | 16-08-2021 11:30 PM |
| FormatDate.dmyDash12WithSec | 16-08-2021 11:30:10 PM |
| FormatDate.mdyDash12 | 08-16-2021 11:30 PM|
| FormatDate.mdyDash12WithSec | 08-16-2021 11:30:10 PM|
| FormatDate.ymdDash12 | 2021-08-16 11:30 PM|
| FormatDate.ymdDash12WithSec | 2021-08-16 11:30:10 PM|
| FormatDate.ydmDash12 | 2021-16-08 11:30 PM|
| FormatDate.ydmDash12WithSec | 2021-16-08 11:30:10 PM|
| FormatDate.dmy12 | 16/08/2021 11:30 PM|
| FormatDate.dmy12WithSec | 16/08/2021 11:30:10 PM|
| FormatDate.mdy12 | 08/16/2021 11:30 PM |
| FormatDate.mdy12WithSec | 08/16/2021 11:30:10 PM |
| FormatDate.ymd12 | 2021/08/16 11:30 PM|
| FormatDate.ymd12WithSec | 2021/08/16 11:30:10 PM |
| FormatDate.ydm12 | 2021/16/08 11:30 PM |
| FormatDate.ydm12WithSec | 2021/16/08 11:30:10 PM |
| FormatDate.dmyDash24 | 16-08-2021 23:30 |
| FormatDate.dmyDash24WithSec | 16-08-2021 23:30:10 |
| FormatDate.mdyDash24 | 08-16-2021 23:30 |
| FormatDate.mdyDash24WithSec | 08-16-2021 23:30:10 |
| FormatDate.ymdDash24 | 2021-08-16 23:30 |
| FormatDate.ymdDash24WithSec | 2021-08-16 23:30:10 |
| FormatDate.ydmDash24 | 2021-16-08 23:30 |
| FormatDate.ydmDash24WithSec | 2021-16-08 23:30:10 |
| FormatDate.dmy24 | 16/08/2021 23:30 |
| FormatDate.dmy24WithSec | 16/08/2021 23:30:10 |
| FormatDate.mdy24 | 08/16/2021 23:30 |
| FormatDate.mdy24WithSec | 08/16/2021 23:30:10 |
| FormatDate.ymd24 | 2021/08/16 23:30 |
| FormatDate.ymd24WithSec | 2021/08/16 23:30:10 |
| FormatDate.ydm24 | 2021/16/08 23:30 |
| FormatDate.ydm24WithSec | 2021/16/08 23:30:10 |
| FormatDate.time12 | 11:30 PM |
| FormatDate.time12WithSec | 11:30:10 PM |
| FormatDate.time24 | 23:30 |
| FormatDate.time24WithSec | 23:30:10 |



CHANGELOG.md
## 0.0.1 [06-08-2021]

* Initial Release.


lib/flutter_string_converter.dart
library flutter_string_converter;

import 'package:intl/intl.dart';

class FormatDate {
  /// example : 16-08-2021
  static final String dmyDash = "dd-MM-yyyy";

  /// example : 08-16-2021
  static final String mdyDash = "MM-dd-yyyy";

  /// example : 2021-08-16
  static final String ymdDash = "yyyy-MM-dd";

  /// example : 2021-16-08
  static final String ydmDash = "yyyy-dd-MM";

  /// example : 16/08/2021
  static final String dmy = "dd/MM/yyyy";

  /// example : 08/16/2021
  static final String mdy = "MM/dd/yyyy";

  /// example : 2021/08/16
  static final String ymd = "yyyy/MM/dd";

  /// example : 2021/16/08
  static final String ydm = "yyyy/dd/MM";

  /// example : 16-08-2021 11:30 PM
  static final String dmyDash12 = "dd-MM-yyyy hh:mm a";

  /// example : 16-08-2021 11:30:23 PM
  static final String dmyDash12WithSec = "dd-MM-yyyy hh:mm:ss a";

  /// example : 08-16-2021 11:30 PM
  static final String mdyDash12 = "MM-dd-yyyy hh:mm a";

  /// example : 08-16-2021 11:30:23 PM
  static final String mdyDash12WithSec = "MM-dd-yyyy hh:mm:ss a";

  /// example : 2021-08-16 11:30 PM
  static final String ymdDash12 = "yyyy-MM-dd hh:mm a";

  /// example : 2021-08-16 11:30:23 PM
  static final String ymdDash12WithSec = "yyyy-MM-dd hh:mm:ss a";

  /// example : 2021-16-08 11:30 PM
  static final String ydmDash12 = "yyyy-dd-MM hh:mm a";

  /// example : 2021-16-08 11:30:23 PM
  static final String ydmDash12WithSec = "yyyy-dd-MM hh:mm:ss a";

  /// example : 16/08/2021 11:30 PM
  static final String dmy12 = "dd/MM/yyyy hh:mm a";

  /// example : 16/08/2021 11:30:23 PM
  static final String dmy12WithSec = "dd/MM/yyyy hh:mm:ss a";

  /// example : 08/16/2021 11:30 PM
  static final String mdy12 = "MM/dd/yyyy hh:mm a";

  /// example : 08/16/2021 11:30:23 PM
  static final String mdy12WithSec = "MM/dd/yyyy hh:mm:ss a";

  /// example : 2021/08/16 11:30 PM
  static final String ymd12 = "yyyy/MM/dd hh:mm a";

  /// example : 2021/08/16 11:30:23 PM
  static final String ymd12WithSec = "yyyy/MM/dd hh:mm:ss a";

  /// example : 2021/16/08 11:30 PM
  static final String ydm12 = "yyyy/dd/MM hh:mm a";

  /// example : 2021/16/08 11:30:23 PM
  static final String ydm12WithSec = "yyyy/dd/MM hh:mm:ss a";

  /// example : 16-08-2021 23:30
  static final String dmyDash24 = "dd-MM-yyyy HH:mm";

  /// example : 16-08-2021 23:30:23
  static final String dmyDash24WithSec = "dd-MM-yyyy HH:mm:ss";

  /// example : 08-16-2021 23:30
  static final String mdyDash24 = "MM-dd-yyyy HH:mm";

  /// example : 08-16-2021 23:30:23
  static final String mdyDash24WithSec = "MM-dd-yyyy HH:mm:ss";

  /// example : 2021-08-16 23:30
  static final String ymdDash24 = "yyyy-MM-dd HH:mm";

  /// example : 2021-08-16 23:30:23
  static final String ymdDash24WithSec = "yyyy-MM-dd HH:mm:ss";

  /// example : 2021-16-08 23:30
  static final String ydmDash24 = "yyyy-dd-MM HH:mm";

  /// example : 2021-16-08 23:30:23
  static final String ydmDash24WithSec = "yyyy-dd-MM HH:mm:ss";

  /// example : 16/08/2021 23:30
  static final String dmy24 = "dd/MM/yyyy HH:mm";

  /// example : 16/08/2021 23:30:23
  static final String dmy24WithSec = "dd/MM/yyyy HH:mm:ss";

  /// example : 08/16/2021 23:30
  static final String mdy24 = "MM/dd/yyyy HH:mm";

  /// example : 08/16/2021 23:30:23
  static final String mdy24WithSec = "MM/dd/yyyy HH:mm:ss";

  /// example : 2021/08/16 23:30
  static final String ymd24 = "yyyy/MM/dd HH:mm";

  /// example : 2021/08/16 23:30:23
  static final String ymd24WithSec = "yyyy/MM/dd HH:mm:ss";

  /// example : 2021/16/08 23:30
  static final String ydm24 = "yyyy/dd/MM HH:mm";

  /// example : 2021/16/08 23:30:23
  static final String ydm24WithSec = "yyyy/dd/MM HH:mm:ss";

  /// example : 11:30 PM
  static final String time12 = "hh:mm a";

  /// example : 11:30:23 PM
  static final String time12WithSec = "hh:mm:ss a";

  /// example : 23:30
  static final String time24 = "HH:mm";

  /// example : 23:30:23
  static final String time24WithSec = "HH:mm:ss";
}

extension StringExtension on String {
  /// for only first letter of string will capital example: "this is a boy".toFirstCapital() => "This is a boy".
  String toFirstCapital() =>
      "${this[0].toUpperCase()}${this.substring(1).toLowerCase()}";

  /// for every first letter of string will capital example: "this is a boy".toEveryFirstCapital() => "This Is A Boy".
  String toEveryFirstCapital() =>
      this.split(" ").map((str) => str.toFirstCapital()).join(" ");

  /// to camel case of string example: "this is a boy".toCamelCase() => "tHiS Is a bOy".
  String toCamelCase() {
    var _str = '';
    for (int i = 0; i < this.length; i++) {
      if (i.isOdd) {
        _str += this[i].toUpperCase();
      } else {
        _str += this[i].toLowerCase();
      }
    }
    return _str;
  }

  /// for use single letter of string example: "Rahul".toSingleLetter() => "R".
  String toSingleLetter() => "${this.replaceAll(this.substring(1), "")}";

  /// for use every single letter of string and join with [joinWith] you pass example: "Rahul Gandhi".toEverySingleLetter(".") => "R.G".
  String toEverySingleLetter({required String joinWith}) =>
      this.split(" ").map((str) => str.toSingleLetter()).join(joinWith);

  /// for convert into date with given format
  String toDate({required String format}) =>
      DateFormat(format).format(DateTime.parse(this)).toString();

  /// to double
  double toDouble() => double.parse(this.toString());

  /// to int
  int toInt() => int.parse(this.toString());

}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_string_converter/flutter_string_converter.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  final String _string = "this is a boy";
  final String _name = "Rahul Gandhi";
  final String _date = "2021-08-16 23:30:23";
  final String _double = "1.2";
  final String _int = "1";


  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Examples'),
      ),
      body: ListView(
        children: [
          Center(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text(
                '$_string.toFirstCapital() => ${_string.toFirstCapital()}',
              ),
            ),
          ),
          Center(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text(
                '$_string.toEveryFirstCapital() => ${_string.toEveryFirstCapital()}',
              ),
            ),
          ),
          Center(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text(
                '$_name.toSingleLetter() => ${_name.toSingleLetter()}',
              ),
            ),
          ),
          Center(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text(
                '${_string.toCamelCase()}',
              ),
            ),
          ),
          Center(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text(
                '${_date.toDate(format: FormatDate.dmyDash24WithSec)}',
              ),
            ),
          ),
          Center(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text(
                '$_double type = ${_double.runtimeType} =>$_double type =  ${_double.toDouble().runtimeType}',
              ),
            ),
          ),
          Center(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text(
                '$_int type = ${_int.runtimeType} =>$_int type =  ${_int.toInt().runtimeType}',
              ),
            ),
          ),
        ],
      ),
    );
  }
}
